uaclient/tests/test_gpg.py

```diff
@@ -70,6 +70,10 @@ def test_user_facing_error_on_invalid_keyring_packet(
 
     def test_export_single_key_from_keyring(self, home_dir, tmpdir):
         """Only a single key is exported from a multi-key source keyring."""
+        if os.getuid() != 0:
+            raise pytest.skip(
+                "Skipping export key test as non-root can't use apt-key"
```

Comment:
Bummer, and it fails even with fakeroot:
```
ubuntu@trusty-ua:~$ rm andreas.gpg 
ubuntu@trusty-ua:~$ apt-key  export 437D05B5 | fakeroot apt-key --keyring andreas.gpg add -
OK
ubuntu@trusty-ua:~$ l andreas.gpg 
-rw-r--r-- 1 ubuntu ubuntu 0 Oct  9 17:23 andreas.gpg
ubuntu@trusty-ua:~$ 
```

Reply:
Oh, fakeroot actually works, I just forgot to use an absolute path for the output keyring file:
```
ubuntu@trusty-ua:~$ rm andreas.gpg*
ubuntu@trusty-ua:~$ apt-key  export 437D05B5 | fakeroot apt-key --keyring ./andreas.gpg add -
OK
ubuntu@trusty-ua:~$ ll andreas.gpg
-rw-r--r-- 1 ubuntu ubuntu 3502 Oct  9 18:39 andreas.gpg
ubuntu@trusty-ua:~$ apt-key --keyring ./andreas.gpg list
./andreas.gpg
-------------
pub   1024D/437D05B5 2004-09-12
uid                  Ubuntu Archive Automatic Signing Key <ftpmaster@ubuntu.com>
sub   2048g/79164387 2004-09-12

ubuntu@trusty-ua:~$ 
```